test/http_downloads_test.rb

```diff
@@ -1,23 +1,41 @@
 require 'test_helper'
+require 'tmpdir'
 
-class Proxy::HttpDownload
-  def command(src, dst)
-    "echo test"
+class HttpDownloadsTest < Test::Unit::TestCase
+  def new_thread_mock
+    thr = mock()
+    thr.stubs(:join).returns(0)
+    thr
   end
-end
 
-class HttpDownloadsTest < Test::Unit::TestCase
-  def test_should_download_a_file
-    assert Proxy::HttpDownloads.start_download 'source', 'destination'
-    assert File.exist?('destination')
+  def tmp(name)
+    File.join(Dir.tmpdir(), "http-download-#{name}.tmp")
+  end
+
+  def test_should_return_an_instance
+    src = tmp('source')
```

Comment:
source path doesn't matter here and in other tests, as it's evaluated by `wget` and no files are being created.